### DISCUSSION TITLE
View Whitelist list

### DISCUSSION THREAD

**Discussion by LeslieDesser** [Cookie-AutoDelete/Cookie-AutoDelete#1109] [Q&A / Support] [answered]
Maybe I'm missing something obvious, but I can't find any way to view the domains I have already whitelisted.

**Comment by kennethtran93** [comment, marked as answer]
You can go into CAD settings either from the popup (last button on top row of buttons) or right clicking on either the extension icon or within a page.

Once inside CAD settings, goto 'List of Expressions' in the navigation on the left to see all configured expressions.

**Reply by LeslieDesser** [reply]
Thank you for the prompt response.

I actually wrote back: I can't see any 'navigation on the left'! .....  and then when wanting to take a snip of the screen I noticed the black strip down the left side!

It is well disguised!  To my eye it merged with the black surround of my screen.  On the right I have my Windows task-bar which is also black, so all I could 'see' was the bright white block in the middle.  Having wide left and right white margins with the options in the middle sort of forces the black strip on the far left into oblivion.

Personal suggestion:  The navigation options would be much more obvious as part of the main white screen.

Many thank!!

**Reply by kennethtran93** [reply]
The navigation was set that way as it would be hidden on mobile view until the hamburger-like menu icon is clicked upon.  Designs are always welcome if you have something in mind that would work in a responsive view.

**Reply by LeslieDesser** [reply]
I think its just that a solid black block on the far edge of the screen is not 'seen' by the eye unless the user consciously looks there.  This is clearly so with the task bar at the bottom of the screen (or in my case on the right).  I do not 'see' it unless looking specifically there.  I suspect it was explicitly designed that way to keep it separate from the working area of the screen.

If it had some border, I think it would more readily be noticed.  If I switch my browser from full-screen to be a window, then the black rectangle becomes immediately 'visible' and actually draws the eye to it.

**Comment by LeslieDesser** [comment]
If I may take the liberty of commenting on the choice of wording. (Having spent 50+ years designing system with usability very much in mind I allow myself such liberties:) ) 

You have Whitelists and Greylists and their meaning are well understood - especially as you actually define them in your notes.

The use of the 'Expressions' really does not conjure up any obvious association with Whitelists and Greylists. Its a technical term that could refer to virtually anything. So even having found the navigation bar, I would still be struggling to find the list I want.

One final suggestion: I think two separate options (List Whitelist; List Greylist) would be more useful than a combined list.  Or else, allow the user to filter the list.

Finally, many thanks for what looks like a very useful Extension.

**Reply by kennethtran93** [reply]
I believe Kenny originally named it as such as we use Regular Expressions.  All domains entered are converted to a regular expression in the backend when parsing the list of cookies.

Albeit not documented properly, you are able to filter by entering the domain in the textbox in that page.

It would be slightly harder to do separate lists due to the fact that we already use firefox's container's as the tabs, so having tabs within a tab may not be the best.

Not to mention that somewhere further down the line, whichever major version it may appear in, the lists may change to one you can customize (i.e. not just white or grey list, and the naming could be customized as well), so further design work needs to be done there.

**Reply by LeslieDesser** [reply]
Understood.  On the subject of using the word 'Expressions', I would still suggest that its use is meaningless to the vast majority of users - whether IT pros or laymen.  It is only by clicking the option to see what it does, did I discover its relationship to White/Grey lists.

**Reply by kennethtran93** [reply]
What would you call it then? This field takes in either a domain or a regular expression that may match multiple domains, or even IP Addresses.

**Reply by LeslieDesser** [reply]
This tool's primary purpose (rule) is to delete all cookies relating to all domains as soon as they are no longer needed.

All the other settings are 'exceptions' to the primary rule.

An 'expression' is the way the designer of this tool has decided to **implement** how to create domain exceptions.

So the list is really a 'Domain Exception List'.   A 'Whitelist' is generally understood to be a list of exceptions to a general rule.  So an alternative could be 'Domain Whitelist'.  The author has chosen to apply the name 'Greylist' to an exception that is time limited.  

So I would suggest using 'Exception List' or 'Domain Exception List' if you plan to add other types of lists, or 'White/Grey List' if you are sticking to just those two.

**Reply by kennethtran93** [reply]
@LeslieDesser Thanks.  I'll take that into consideration for the next major version.
